File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/presentation/ui/transaksi_penjualan/TransaksiPenjualanAdapter.java
package com.bits.bee.bpmc.presentation.ui.transaksi_penjualan;

import java.lang.System;

/**
 * Created by aldi on 23/06/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u00006\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0010\b\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0004\u0018\u00002\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00030\u0001:\u0002\u0012\u0013B\u0019\u0012\u0012\u0010\u0004\u001a\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00060\u0005\u00a2\u0006\u0002\u0010\u0007J\u0018\u0010\n\u001a\u00020\u00062\u0006\u0010\u000b\u001a\u00020\u00032\u0006\u0010\f\u001a\u00020\rH\u0016J\u0018\u0010\u000e\u001a\u00020\u00032\u0006\u0010\u000f\u001a\u00020\u00102\u0006\u0010\u0011\u001a\u00020\rH\u0016R\u000e\u0010\b\u001a\u00020\tX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u001a\u0010\u0004\u001a\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00060\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0014"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/transaksi_penjualan/TransaksiPenjualanAdapter;", "Landroidx/paging/PagingDataAdapter;", "Lcom/bits/bee/bpmc/domain/model/Sale;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "onItemClick", "Lkotlin/Function1;", "", "(Lkotlin/jvm/functions/Function1;)V", "hourFormat", "Ljava/text/SimpleDateFormat;", "onBindViewHolder", "holder", "position", "", "onCreateViewHolder", "parent", "Landroid/view/ViewGroup;", "viewType", "DiffCallback", "ViewHolder", "app_debug"})
public final class TransaksiPenjualanAdapter extends androidx.paging.PagingDataAdapter<com.bits.bee.bpmc.domain.model.Sale, androidx.recyclerview.widget.RecyclerView.ViewHolder> {
    private final kotlin.jvm.functions.Function1<com.bits.bee.bpmc.domain.model.Sale, kotlin.Unit> onItemClick = null;
    private final java.text.SimpleDateFormat hourFormat = null;
    
    public TransaksiPenjualanAdapter(@org.jetbrains.annotations.NotNull()
    kotlin.jvm.functions.Function1<? super com.bits.bee.bpmc.domain.model.Sale, kotlin.Unit> onItemClick) {
        super(null, null, null);
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public androidx.recyclerview.widget.RecyclerView.ViewHolder onCreateViewHolder(@org.jetbrains.annotations.NotNull()
    android.view.ViewGroup parent, int viewType) {
        return null;
    }
    
    @java.lang.Override()
    public void onBindViewHolder(@org.jetbrains.annotations.NotNull()
    androidx.recyclerview.widget.RecyclerView.ViewHolder holder, int position) {
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u001e\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\b\u0086\u0004\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u000e\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\bR\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\t"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/transaksi_penjualan/TransaksiPenjualanAdapter$ViewHolder;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "binding", "Lcom/bits/bee/bpmc/databinding/ItemTransaksiPenjualanBinding;", "(Lcom/bits/bee/bpmc/presentation/ui/transaksi_penjualan/TransaksiPenjualanAdapter;Lcom/bits/bee/bpmc/databinding/ItemTransaksiPenjualanBinding;)V", "bind", "", "model", "Lcom/bits/bee/bpmc/domain/model/Sale;", "app_debug"})
    public final class ViewHolder extends androidx.recyclerview.widget.RecyclerView.ViewHolder {
        private final com.bits.bee.bpmc.databinding.ItemTransaksiPenjualanBinding binding = null;
        
        public ViewHolder(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.databinding.ItemTransaksiPenjualanBinding binding) {
            super(null);
        }
        
        public final void bind(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Sale model) {
        }
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u0018\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u000b\n\u0002\b\u0004\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0003J\u0018\u0010\u0004\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00022\u0006\u0010\u0007\u001a\u00020\u0002H\u0016J\u0018\u0010\b\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00022\u0006\u0010\u0007\u001a\u00020\u0002H\u0016\u00a8\u0006\t"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/transaksi_penjualan/TransaksiPenjualanAdapter$DiffCallback;", "Landroidx/recyclerview/widget/DiffUtil$ItemCallback;", "Lcom/bits/bee/bpmc/domain/model/Sale;", "()V", "areContentsTheSame", "", "oldItem", "newItem", "areItemsTheSame", "app_debug"})
    public static final class DiffCallback extends androidx.recyclerview.widget.DiffUtil.ItemCallback<com.bits.bee.bpmc.domain.model.Sale> {
        
        public DiffCallback() {
            super();
        }
        
        @java.lang.Override()
        public boolean areItemsTheSame(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Sale oldItem, @org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Sale newItem) {
            return false;
        }
        
        @java.lang.Override()
        public boolean areContentsTheSame(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Sale oldItem, @org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Sale newItem) {
            return false;
        }
    }
}

File: app/src/main/java/com/bits/bee/bpmc/presentation/ui/transaksi_penjualan/TransaksiPenjualanAdapter.kt
package com.bits.bee.bpmc.presentation.ui.transaksi_penjualan

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.core.view.isVisible
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.bits.bee.bpmc.R
import com.bits.bee.bpmc.databinding.ItemHeaderBinding
import com.bits.bee.bpmc.databinding.ItemTransaksiPenjualanBinding
import com.bits.bee.bpmc.domain.model.Sale
import com.bits.bee.bpmc.utils.BPMConstants
import com.bits.bee.bpmc.utils.BeePreferenceManager
import com.bits.bee.bpmc.utils.CurrencyUtils
import com.bits.bee.bpmc.utils.DateFormatUtils
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by aldi on 23/06/22.
 */
class TransaksiPenjualanAdapter(
    private val onItemClick : (Sale) -> Unit,
    private var activeSale : Sale?,
) : PagingDataAdapter<Sale, RecyclerView.ViewHolder>(DiffCallback()){

    private val hourFormat = SimpleDateFormat("HH:mm", Locale.getDefault())

    fun setActiveSale(sale : Sale?){
        this.activeSale = sale
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return if(viewType == 1)
            ViewHolder(
                ItemTransaksiPenjualanBinding.inflate(inflater, parent, false)
            )
        else
            HeaderViewHolder(
                ItemHeaderBinding.inflate(inflater, parent, false)
            )
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if(holder.itemViewType == 1) {
            val viewHolder = holder as ViewHolder
            getItem(position)?.let {
                viewHolder.bind(it)
            }
        } else {
            val viewHolder = holder as HeaderViewHolder
            getItem(position)?.let {
                viewHolder.bind(it)
            }
        }
    }

    override fun getItemViewType(position: Int): Int {
        return if(getItem(position)?.id == null) 0 else 1
    }

    inner class ViewHolder(private val binding : ItemTransaksiPenjualanBinding) : RecyclerView.ViewHolder(binding.root){
        fun bind(model : Sale){
            binding.apply {

                activeSale?.let {
                    if(BeePreferenceManager.ORIENTATION == BPMConstants.SCREEN_LANDSCAPE && model.id == it.id)
                        clContent.setBackgroundColor(ContextCompat.getColor(binding.root.context, R.color.light_red))
                    else
                        clContent.setBackgroundColor(ContextCompat.getColor(binding.root.context, R.color.white))
                }

                tvNamaCust.text = model.bpName

                tvTotal.text = CurrencyUtils.formatCurrency(model.total)

                tvChannel.text = model.channel

                tvJam.text = hourFormat.format(model.trxDate)

                cvVoid.isVisible = model.isVoid

                clContent.setOnClickListener {
                    onItemClick(model)
                }
            }
        }
    }

    inner class HeaderViewHolder(private val binding : ItemHeaderBinding) : RecyclerView.ViewHolder(binding.root){
        fun bind(model : Sale){
            binding.apply {
                tvTglSesi.text = DateFormatUtils.getDate(model.trxDate)
            }
        }
    }

    class DiffCallback : DiffUtil.ItemCallback<Sale>(){
        override fun areItemsTheSame(oldItem: Sale, newItem: Sale): Boolean {
            return oldItem == newItem
        }

        override fun areContentsTheSame(oldItem: Sale, newItem: Sale): Boolean {
            return oldItem.id == newItem.id
        }
    }
}